src/canopy_bioemi_mod.F90

```diff
@@ -298,15 +305,89 @@ SUBROUTINE CANOPY_BIO( ZK, FCLAI, FCH, LAI, CLU, COSZEN, SFCRAD, &
 
         GammaPPFD_AVE = (GammaPPFD_SUN*FSUN) + (GammaPPFD_SHADE*(1.0-FSUN)) ! average = sum sun and shade weighted by sunlit fraction
 
+! Get CO2 inhibition factor for isoprene only
+
+        if (EMI_IND .eq. 1) then  !Isoprene
+            GAMMACO2 = GET_GAMMA_CO2(CO2OPT,CO2SET)
+        else
+            GAMMACO2 = 1.0_rk
+        end if
+
 ! Calculate emissions profile in the canopy
         EMI_OUT = 0.0_rk  ! set initial emissions profile to zero
-        do i=1, SIZE(ZK)
-            if (ZK(i) .gt. 0.0 .and. ZK(i) .le. FCH) then  ! above ground level and at/below canopy top
-                FLAI(i) = ((FCLAI(i+1) - FCLAI(i)) * LAI)/MODRES    !fractional LAI in each layer converted to LAD (m2 m-3)
-                EMI_OUT(i) = FLAI(i) * EF * GammaTLEAF_AVE(i) * GammaPPFD_AVE(i) * CCE  ! (ug m-3 hr-1)
-                EMI_OUT(i) = EMI_OUT(i) * 2.7777777777778E-13_rk !TBD:  convert emissions output to (kg m-3 s-1)
-            end if
-        end do
+        FLAI = 0.0_rk  ! set initial fractional FLAI (LAD) profile to zero
+
+        if (VERT .eq. 0) then         !Full 3D leaf-level biogenic emissions (no averaging, summing, or integration)
+            do i=1, SIZE(ZK)
+                if (ZK(i) .gt. 0.0 .and. ZK(i) .le. FCH) then           ! above ground level and at/below canopy top
+                    FLAI(i) = ((FCLAI(i+1) - FCLAI(i)) * LAI)/MODRES    !fractional LAI in each layer converted to LAD (m2 m-3)
+                    EMI_OUT(i) = FLAI(i) * EF * GammaTLEAF_AVE(i) * GammaPPFD_AVE(i) * GAMMACO2 * CCE  ! (ug m-3 hr-1)
+                    EMI_OUT(i) = EMI_OUT(i) * 2.7777777777778E-13_rk    !convert emissions output to (kg m-3 s-1)
+                end if
+            end do
+        else if (VERT .eq. 1) then       !"MEGANv3-like": Use weighting factors normalized to plant distribution shape (FCLAI)
+            !across canopy layers
+            LAYERS = floor(FCH/MODRES)
```

Comment:
```suggestion
            LAYERS = floor(FCH/MODRES) + 1
```
My testing suggests need +1 here. This makes the results consistent with canopy integrating the layerwise emission results.

Example: hc is 27.8, res is 0.5. floor(hc / res) is 55. But 0:0.5:27.5 (canopy layer heights) has length 56.

Reply:
I agree Zach and have committed suggestion.

Reply:
@zmoon Would also +1 be needed in the ``` LAYERS = floor(FCH/MODRES)  ``` for VERT options 2 and 3 then as well?

Reply:
Yes I believe so, I think I checked option 2 as well and VPGWT didn't sum to 1 currently.

Reply:
OK, I thought I had checked that and they all summed to 1 OK without the +1.   

sum(VPGWT(1:LAYERS)):
...
1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
   1.0000000000000000
...

Reply:
I add the layers+1 to all vert options [here](https://github.com/noaa-oar-arl/canopy-app/pull/73/commits/1c6481f7bb78b4bf9b797e769a14de530f4ca54b).